posts/2020-11-27-typing-your-actors.adoc

```diff
@@ -0,0 +1,387 @@
+= Typing your actors
+pedrolunatech
+v1.0, 2020-11-27
+:title: Typing your actors step by step
+:tags: [akka, migration, akka-typed]
+
+== Background
+
+After some frustration with our project's choice for planning poker solution, I did what any reasonable developer would do. No, I didn't search for a more suitable solution, I decided to write mine and do a blog post about it.
+
+The main idea to build a simple application that uses some features of the following core Akka libraries:
+
+* https://doc.akka.io/docs/akka-http/current/index.html[Akka HTTP]:  REST API for Vue.js frontend
+* https://doc.akka.io/docs/akka/current/stream/index.html[Akka Streams]: Websocket bi-directional data handling
+* https://doc.akka.io/docs/akka/current/typed/index.html[Akka Actors]: Rooms management
+
+Initially, I deliberately chose to use a classic actor implementation so that I could actually also talk about something that might be more useful: migrating an Akka Classic application to an Akka Typed version.
+
+Why would you want to do that, well for me it is the type safety of not having everything being a function of `Any => Unit` **_and_** strong support to avoid anti-patterns (more eloquently explained https://www.youtube.com/watch?v=vF_-28I6yGE[here]), but feel free to search more if this migration is something for your project.
```

Comment:
This change is a mix between what was there and what I proposed. The bit before the comma is a question, which isn't obvious when you look at it.

Reply:
Yeap, fixing it